File: Python/beginner/1435.py
# https://judge.beecrowd.com/pt/problems/view/1435
# Matriz Quadrada I

while True: # This will iterate until n = 0
    
    # Getting the size of the matrix
    n = int(input())
    
    if n == 0: # Testing if n = 0
        break # Breaking the loop if n = 0
    
    # Printing the matrix
    for i in range(n):
        
        line_elements = [] # To store the elements of each line
        
        for j in range(n):
            # Finding the value of each element        
            elementValue = (min(min(i, n - i - 1), min(j, n - j - 1))) + 1 # Here We need to Add 1 because the index starts at 0
            
            if j == 0:
                line_elements.append(f'{elementValue:3d}') # Adding the element to the list with 3 digits
            else:
                line_elements.append(f'{elementValue:4d}') # Adding the element to the list with 4 digits
        
        # Printing the line
        print(''.join(line_elements))
        
    print() # Adding an empty line
        
        
# By: Fernando Serra
# https://github.com/fernandosserra

File: C/beginner/1435.c
// https://judge.beecrowd.com/pt/problems/view/1435
// Matriz Quadrada I

// Libraries
#include <stdio.h>

// Função auxiliar para encontrar o mínimo entre dois números, como no código de referência
int min(int a, int b)
{
    return a < b ? a : b;
}

// Main function
int main()
{
    int n;

    while (scanf("%d", &n) != EOF && n != 0) // Continue reading until EOF (end of file) or 0
    {
        for (int i = 0; i < n; i++) // Iterate between rows
        {
            for (int j = 0; j < n; j++) // Iterate between columns
            {
                // Logic to find the value of each element
                int value = min(min(i, n - 1 - i), min(j, n - 1 - j)) + 1;

                // This fucking format is the key of this problem
                if (j == 0) { // If is the first element of the line
                    printf("%3d", value); // print with 3 digits (ex: " 1", "10")
                } else { // For the other elements
                    printf("%4d", value); // print with 4 digits (ex: " 1  ", "10  ")
                }
            }
            printf("\n"); // Breaking the LAWne
        }
        printf("\n"); // With line after each matrix
        
    }

    return 0;
}

// By Fernando Serra (Most of the code is built with help of Washu-sama)
// https://github.com/fernandosserra